### DISCUSSION TITLE
Query builder methods for generic handlers

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#179] [Ideas]
Sometimes it's useful for narrowing a query set in generic handlers.
Take for example a `?search=...` query. The `RecordList` generic handler would be sufficient if there was a way to narrow down the query. Also, it would be very useful for implementing soft delete.

```crystal
class ArticleListHandler < Marten::Handlers::RecordList
  model Article
  template_name "article_list.html"

  # The generic handler then adds its own filters on the result of this
  def query
    model.filter(title__contains: request.query_params["search"])
  end
end
```

**Comment by notramo** [comment]
Oh, I found it, it's `queryset`. However, it would be nice to support that macro for `RecordDetail` and `RecordUpdate` too.

**Comment by ellmetha** [comment]
I agree, all handlers that use a query set for retrieving records should have this macro. This should be an easy addition to the framework.
